agent/containers_dockercli.go

```diff
@@ -0,0 +1,188 @@
+package agent
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coder/coder/v2/codersdk"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/xerrors"
+)
+
+// dockerCLIContainerLister is a ContainerLister that lists containers using the docker CLI
+type dockerCLIContainerLister struct{}
+
+var _ ContainerLister = &dockerCLIContainerLister{}
+
+func (*dockerCLIContainerLister) List(ctx context.Context) ([]codersdk.WorkspaceAgentContainer, error) {
+	var buf bytes.Buffer
+	// List all container IDs, one per line, with no truncation
+	cmd := exec.CommandContext(ctx, "docker", "ps", "--all", "--quiet", "--no-trunc")
```

Comment:
Does this work across all versions of docker?

Reply:
Thanks for going back a few versions to check 👍 

I think we can leave it for now. If we find this causes issues later we can refactor the args into version types.